packages/bsb-core/bsb/simulation/adapter.py

```diff
@@ -129,8 +185,25 @@ def collect(self, results):
             result.flush()
         return results
 
-    def add_progress_listener(self, listener):
-        self._progress_listeners.append(listener)
-
-
-__all__ = ["AdapterProgress", "SimulationData", "SimulatorAdapter"]
+    def implement_components(self, simulation):
+        simdata = self.simdata[simulation]
+        for component in simulation.get_components():
+            component.implement(self, simulation, simdata)
+
+    def load_controllers(self, simulation):
+        for component in simulation.get_components():
+            if hasattr(component, "get_next_checkpoint"):
+                if hasattr(component, "progress"):
```

Comment:
Is this the `run_checkpoint` hook? Why was it renamed to `progress`? I think it's sort of unclear, each piece of this feature should be clearly semantically linked to "checkpoints".